enhancements/cloud-integration/openstack/failure-domains.md

```diff
@@ -0,0 +1,720 @@
+
+Start by filling out the header with the metadata for this enhancement.
+
+# OpenStack Failure Domains
+
+## Summary
+
+OpenStack deployments do not have a single implementation of failure domains,
+but instead define availability zones individually for each service, e.g. Nova
+(compute), Neutron (network), and Cinder (storage). OpenShift is capable of
+deploying its control plane across multiple failure domains, but for OpenStack
+this is currently limited to compute availability zones. This enhancement
+proposes a unified definition of failure domains for an OpenStack deployment
+covering all supported availability zone types, and defines how OpenShift will
+use them when deploying the control plane.
+
+An example deployment with 3 failure domains is shown below:
+
+```mermaid
+graph BT
+    subgraph Spine
+        spine[Spine]
+    end
+ 
+    subgraph Failure Domain 1
+        direction TB
+   
+        control_plane_net_1(Control plane subnet 1)
+        storage_net_1(Storage subnet 1)
+ 
+        subgraph Nova AZ 1
+            control_1(Control plane node)
+        end
+ 
+        subgraph Cinder AZ 1
+            storage_1(Cinder storage)
+        end
+ 
+        control_1 -- tag: control_plane --- control_plane_net_1
+        control_1 -- tag: storage --- storage_net_1
+        storage_1 --- storage_net_1
+    end
+ 
+    subgraph Failure Domain 2
+        direction TB
+   
+        control_plane_net_2(Control plane subnet 2)
+        storage_net_2(Storage subnet 2)
+ 
+        subgraph Nova AZ 2
+            control_2(Control plane node)
+        end
+ 
+        subgraph Cinder AZ 2
+            storage_2(Cinder storage)
+        end
+ 
+        control_2 -- tag: control_plane --- control_plane_net_2
+        control_2 -- tag: storage --- storage_net_2
+        storage_2 --- storage_net_2
+    end
+ 
+    subgraph Failure Domain 3
+        direction TB
+   
+        control_plane_net_3(Control plane subnet 3)
+        storage_net_3(Storage subnet 3)
+ 
+        subgraph Nova AZ 3
+            control_3(Control plane node)
+        end
+ 
+        subgraph Cinder AZ 3
+            storage_3(Cinder storage)
+        end
+ 
+        control_3 -- tag: control_plane --- control_plane_net_3
+        control_3 -- tag: storage --- storage_net_3
+        storage_3 --- storage_net_3
+    end
+ 
+    control_plane_net_1 --- spine
+    control_plane_net_2 --- spine
+    control_plane_net_3 --- spine
+```
+
+## Motivation
+
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+
+### User Stories
+
+* As an administrator, I want to ensure my control plane is resilient to the
+  failure of a single compute availability zone.
+* As an administrator, I want to ensure my control plane is resilient to the
+  failure of a single storage availability zone.
+* As an administrator, I want to deploy my control plane across multiple
```

Comment:
This user story doesn't have a why, just a what, why does deploying across multiple networks help the administrator